Assets/Scripts/Models/FurnitureJobs.cs

```diff
@@ -0,0 +1,214 @@
+    /// </summary>
+    /// <value>The spot offset where the Character will stand when he uses the furniture.</value>
+    public Vector2 WorkSpotOffset { get; private set; }
+
+    /// <summary>
+    /// Gets the spot offset where inventory is spawn when a Job is done with this machine.
+    /// </summary>
+    /// <value>The spawn spot offset.</value>
+    public Vector2 SpawnSpotOffset { get; private set; }
+
+    /// <summary>
+    /// Gets the <see cref="Job"/> with the specified index.
+    /// </summary>
+    /// <param name="i">The index.</param>
+    public Job this[int i]
+    {
+        get
+        {
+            return activeJobs[i];
+        }
+    }
+
+    /// <summary>
+    /// Gets the tile that is used to do a job.
+    /// </summary>
+    /// <returns>Tile that is used for jobs.</returns>
+    public Tile GetWorkSpotTile()
+    {
+        return World.Current.GetTileAt(furniture.Tile.X + (int)WorkSpotOffset.x, furniture.Tile.Y + (int)WorkSpotOffset.y, furniture.Tile.Z);
+    }
+
+    /// <summary>
+    /// Gets the tile that is used to spawn new objects (i.e. Inventory, Character).
+    /// </summary>
+    /// <returns>Tile that is used to spawn objects (i.e. Inventory, Character).</returns>
+    public Tile GetSpawnSpotTile()
+    {
+        return World.Current.GetTileAt(furniture.Tile.X + (int)SpawnSpotOffset.x, furniture.Tile.Y + (int)SpawnSpotOffset.y, furniture.Tile.Z);
+    }
+
+    /// <summary>
+    /// Reads the job work spot offset from the xml.
+    /// </summary>
+    /// <param name="reader">The Xml Reader.</param>
+    public void ReadWorkSpotOffset(XmlReader reader)
+    {
+        WorkSpotOffset = new Vector2(
+            int.Parse(reader.GetAttribute("X")),
+            int.Parse(reader.GetAttribute("Y")));
+    }
+
+    /// <summary>
+    /// Reads the spawn spot offset from the xml.
+    /// </summary>
+    /// <param name="reader">The Xml Reader.</param>
+    public void ReadSpawnSpotOffset(XmlReader reader)
+    {
+        SpawnSpotOffset = new Vector2(
+            int.Parse(reader.GetAttribute("X")),
+            int.Parse(reader.GetAttribute("Y")));
+    }
+
+    /// <summary>
+    /// How many active jobs are linked to this furniture.
+    /// </summary>
+    /// <returns>The number of jobs linked to this furniture.</returns>
+    public int Count()
+    {
+        return activeJobs.Count;
+    }
+
+    /// <summary>
+    /// Link a job to the current furniture.
+    /// </summary>
+    /// <param name="job">The job that you want to link to the furniture.</param>
+    public void Add(Job job)
+    {
+        job.furniture = furniture;
+        activeJobs.Add(job);
+        job.OnJobStopped += OnJobStopped;
+        World.Current.jobQueue.Enqueue(job);
+    }
+
+    /// <summary>
+    /// Cancel all the jobs linked to the current furniture.
+    /// </summary>
+    public void CancelAll()
+    {
+        Job[] jobsArray = activeJobs.ToArray();
```

Comment:
All the paused jobs are already canceled. For now pause cancels a job to remove them from the queue and the character, and stores them in a list, to save the progress. Later pause adds all the jobs to the jobs queue, like if it was a new job, but it has the same progress as before.